recipe/meta.yaml

```diff
@@ -25,15 +24,16 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
   host:
-    - {{ mpi }}
+    - mpi
     - python
-    - swig 3.0.10
+    - swig 3.0.11
     - doxygen
     - numpy
     - mpi4py
+    - trilinos {{ version }}
```

Comment:
Are you sure you are using the prebuilt trilinos package? Compilation seems to be taking a while.

Reply:
Thanks for all the help.  The PyTrilinos and Trilinos feedstocks haven't been updated in several months, and there is plenty of bit rot to address.  Some of what you have attempted to fix was just me trying to isolate issues, but that's fine.

Unfortunately, there isn't a way to build PyTrilinos using an installed version of Trilinos.  My discussions with conda-forge developers led me to believe that things would still work.

Reply:
so do you need to depend on this package at all? will you be creating a conflict between packages?

Reply:
We'll see.  This is the first time I am trying it.

Actually, I have been seeing a compilation error that might be due to a conflict between the Trilinos and PyTrilinos feedstocks ... it's been bugging me, but if the PyTrilinos build system is pulling in the installed Trilinos header (and configuration) files, then it would explain the compilation error.

Perhaps I need to set up Trilinos and PyTrilinos to be mutually exclusive.  You can install one but not both.  If you install PyTrilinos, it will have everything Trilinos has, plus the python wrappers.  Can we do that?

Reply:
I kinda did that for opencv.

You have one package install everything. But the other add in the python dependencies.

It will polute the environment with a few  `site-packages` directories, for those that do not want python, but it avoids conflicts. First lets try to get the package to compile. You got really close!
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=107389&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=6d4b912b-175d-51da-0fd9-4d30fe1eb4e7&l=4646

Reply:
Yes, it did get close.  And on Linux, we do not see the compilation error I was referencing.

How do you propose we proceed?  If it was just me, I might try reproducing the "no constructor" error locally using docker, although while I have installed docker locally, I haven't really ever used it.

I'm pretty sure we could disable almost all the Trilinos packages except Teuchos and still reproduce the error.  This would save a lot of time.

Reply:
I would apply a few of the cleanups i recommended, and stop skipping python3 or osx. 12.18.1 is causing me griefs.

Reply:
OK.  I'll apply your cleanups.  I was skipping stuff to isolate problems.

The reason I was working with 12.12.1 was to see if I could set up the dependencies issue properly without the headache of newer versions.

I have gotten the current version of Trilinos/PyTrilinos to compile locally under an Anaconda environment to supply python, numpy, swig, etc.  I forget if there is a 12.18.1 tarball, but if there is, my changes have probably not made it into it yet.

Reply:
Yeah, i've been trying to patchup 12.18.1 with what seems to be yoru changes. Either will work.

Reply:
Are you trying to create a 12.18.1 feedstock, or just get PyTrilinos to compile locally under Trilinos 12.18.1?

Reply:
See my effort here: https://github.com/conda-forge/pytrilinos-feedstock/pull/22